### ISSUE TITLE
Бан аккаунта

### ISSUE THREAD

**Issue by Nikita2305** [python273/vk_api#432]
Изначально аккаунт не заблокирован.

После авторизации, а затем отправки первого запроса получаю ошибку:
[5] User authorization failed: user is blocked.

Затем оказывается, что аккаунт заблокирован.
Аккаунт можно восстановить, но не более 2-3 раз.

Ошибка возникает при отправке запроса с конкретного ПК, с которого ранее(несколько дней назад) было осуществлено много запросов к API. На другом ПК ошибка не воспроизвелась.
Что может быть причиной?

### Окружение

vk_api: 11.9.6

### Пример

```python
vk_session = vk_api.VkApi(login="89131436601", password="ama_Red_god")
vk_session.auth()
vk = vk_session.get_api()

link = input()
link = link.split('/')[-1]
vk.users.get(user_ids=link)
```

**Comment by python273** [comment]
Не уверен в чем именно причина, другие люди тоже писали о блокировках.

Если нужна аутентификация по логину и паролю, то попробуйте создать свое приложение и передавать в `app_id`. Если не нужна, то руками получить токен и передавать в `token`.

**Comment by sadspirit** [comment]
По результатам экспериментов (20 пожженых аккаунтов):
Банилка реагирует на флоу авторизации (на что конкретно - не знаю). Либо сразу блочит, либо пропускает и блочит через несколько минут.

Точно не помогает:
- смена appid (свои/чужие/новые/старые)
- смена IP

Если юзать токен полученный легальным путем и не трогать auth, то все работает (в том числе с скомпрометированных 
 многочисленными банами ip)

**Comment by python273** [comment]
Можно еще попробовать ставить другой юзерагент

например
```python
vk_session.http.headers['User-agent'] = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:94.0) Gecko/20100101 Firefox/94.0'
```

**Comment by sadspirit** [comment]
Да, юзер агент тоже не помогает. Пробовал разные варианты. Также менял версию API. Выставлял задержку между запросами в секунду.
Особенности блокировки:
- Блокировка: Подозрительная активность.
- Токен получает нормально, и уже на 1-10 запросе блочит.
- Блок прилетает на методы group/wall. К примеру на перечисление городов (список стран/городов) - не трогает

Учитывая, вышеперечисленное, думаю аккаунт попадает в подозрительные, после прохождения flow авторизации. Может не хватает какого параметра при эмуляции входа или js чек какой-нибудь вставили (не рисерчил).

**Comment by zt50tz** [comment]
@sadspirit 
> Если юзать токен полученный легальным путем

Прошу прощения, можете раскрыть тему про "легальный путь" — как получить такой токен?

**Comment by Ne0shka** [comment]
Аналогичная проблема - блокнули страницу 2 раза подряд. Рецепт: залогиниться и выполнить users.get. Запрос выполняют и блочат страницу. Вычисляют ботов?

**Comment by DmitryZoloto** [comment]
> @sadspirit
> 
> > Если юзать токен полученный легальным путем
> 
> Прошу прощения, можете раскрыть тему про "легальный путь" — как получить такой токен?
> 
> Имеется ввиду создание приложения и получение токена через него?

https://vk.com/dev/first_guide

**Comment by r35p3ct** [comment]
Подтвержаю, с позавчерашнего дня стали банить.

**Comment by yinelos** [comment]
Друзья, если найдете решение данной проблемы - отпишите пожалуйста сюда.

**Comment by l1zt0p** [comment]
Дня 3 назад улетел первый парсер у меня, окей, вставляю другой аккаут, вчера ночью замечаю, что он в бане, разбаниваю, потом смотрю, что он отлетел вновь сразу после авторизации. Тоже буду копать в чем дело, видимо они решили жестко вставить палки в колеса ботоводам.

**Comment by r35p3ct** [comment]
> Друзья, если найдете решение данной проблемы - отпишите пожалуйста сюда.

Получил токен через [https://oauth.vk.com/authorize?client_id=приложение&display=page&redirect_uri=https://oauth.vk.com/blank.html&response_type=token
&scope=wall,offline](url) ручками и передал его:
`vk_session = vk_api.VkApi(token=acc_token)`

**Comment by dashedman** [comment]
А получить remixsid возможно только через авторизацию по логину?

**Comment by rasperepodvipodvert** [comment]
> Получил токен через https://oauth.vk.com/authorize?client_id=приложение&display=page&redirect_uri=https://oauth.vk.com/blank.html&response_type=token
&scope=wall,offline ручками и передал его:
vk_session = vk_api.VkApi(token=acc_token)

>еще добавил, на всякий случай:
vk_session.http.headers[ 'User-agent'] = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:94.0) Gecko/20100101 Firefox/94.0'

Попробовал выше описанные примеры - либо бан либо требует 'Login is required to auth'
![изображение](https://user-images.githubusercontent.com/14201321/142690323-6cd0dc3e-81fd-4d43-9cd1-12bf5b0b6268.png)

**Comment by vypivshiy** [comment]
При реверсе их официального приложения ВК было выявлено, что можно авторизовываться через bearer, подставляя токен в значение заголовка. 
Брал токены от разных приложений (kate, vk admin etc) и работает. (Брал руками токен, на баны не проверял, проверьте и отпишитесь)

```python
from requests import Session
import vk_api

user_agent = {"User-Agent": "VKShitApp/6.9.1-5828 (Android 14.8.8; SDK 26; armeabi-v7a; foobar ShitVkOS; ru; 800x640)"}

token = "my_secret_token"
session = Session()
session.headers.update(user_agent)
session.headers.update({"Authorization": f"Bearer {token}"})

vk_session = vk_api.VkApi(session=session, api_version="5.181")
vk_api_session = vk_session.get_api()
print(vk_api_session.account.getInfo())
```

**Comment by l1zt0p** [comment]
За метод выше блокировка по подозрительной попытке авторизации сразу.
Попробую поковыряться сейчас, сделать чуть другой метод.

**Comment by rasperepodvipodvert** [comment]
Удалось починить вот таким методом, с предварительным удалением `vk_config.v2.json`

```python
vk_session = vk_api.VkApi(
            login=settings.VK_LOGIN,
            password=settings.VK_PASSWORD,
        )
        try:
            vk_session.http.headers['User-agent'] = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0'
            vk_session.auth(token_only=True)
        except vk_api.AuthError as error_msg:
            print(error_msg)
            return error_msg
```

**Comment by dashedman** [comment]
> Удалось починить вот таким методом, с предварительным удалением `vk_config.v2.json`
> 
> ```python
> vk_session = vk_api.VkApi(
>             login=settings.VK_LOGIN,
>             password=settings.VK_PASSWORD,
>         )
>         try:
>             vk_session.http.headers['User-agent'] = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0'
>             vk_session.auth(token_only=True)
>         except vk_api.AuthError as error_msg:
>             print(error_msg)
>             return error_msg
> ```

а без token_only=True не работает?

**Comment by MrTransistorsChannel** [comment]
Может быть связано с #437

**Comment by salykin** [comment]
В моём случае сработало так:
```py
vk_session = vk_api.VkApi(login, password)
vk_session.http.headers['User-agent'] = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0'
vk_session.auth()
```
Не уверен, может быть, дело было в аккаунтах.

**Comment by AsmanovLev** [comment]
Здравствуйте, может быть вам поможет аутификация с токеном? Я вот просто сколько уже сижу - ни разу не банили.
```# -*- coding: utf-8 -*-
import vk_api
import urllib3
urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
from vk_api.longpoll import VkLongPoll, VkEventType
banlist = [000000000,000000000] # тут писать id-шники тех кого хочешь удалять сообщения 
def auth_handler():
    key = input("Введите код для двухфакторной аутификации: ")
    remember_device = True
    return key, remember_device
def main():                       
    login, password = '+78005553535','password'
    vk_session = vk_api.VkApi(
        login, password, token="0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef01234", app_id="2685278",
        auth_handler=auth_handler
    )
    try:
        vk_session.auth()
    except vk_api.AuthError as error_msg:
        print(error_msg)
        return

    longpoll = VkLongPoll(vk_session)
    vk = vk_session.get_api()
    for event in longpoll.listen():
        if event.type == VkEventType.MESSAGE_NEW:
            print('New message:')
            if event.from_user or event.from_chat:
                if event.user_id in banlist:
                    vk.messages.delete(message_ids=event.message_id, peer_id=event.peer_id)
                if event.from_chat:
                    print(event.user_id, 'in chat', event.chat_id)
                elif event.from_user:
                    print('from user:', event.user_id)
            print('Text: ', event.text)
            print()


if __name__ == '__main__':
    main()
```
Токен получить можно [тут](https://vkhost.github.io/). В примере указан app_id для kate mobile

**Comment by mewforest** [comment]
> Здравствуйте, может быть вам поможет аутификация с токеном? Я вот просто сколько уже сижу - ни разу не банили.

Большое человеческое СПАСИБО! 
Наконец-то получилось пофиксить вход для проблемных аккаунтов, над которой бился несколько суток.
